packages/contracts/test/contracts/ovm/decompression/SequencerMessageDecompressor.spec.ts

```diff
@@ -0,0 +1,333 @@
+import { expect } from '../../../setup'
+
+/* External Imports */
+import { ethers } from '@nomiclabs/buidler'
+import { ContractFactory, Wallet, UnsignedTransaction, Contract } from 'ethers'
+import { zeroPad } from '@ethersproject/bytes'
+import {
+  remove0x,
+  numberToHexString,
+  hexStrToBuf,
+} from '@eth-optimism/core-utils'
+
+/* Internal Imports */
+import {
+  AddressResolverMapping,
+  makeAddressResolver,
+  getWallets,
+  ZERO_ADDRESS,
+  getSignedComponents,
+  getRawSignedComponents,
+  deployAndRegister,
+  getDefaultGasMeterParams,
+  manuallyDeployOvmContract,
+  executeTransaction,
+} from '../../../test-helpers'
+
+const serializeEthSignTransaction = (transaction: any): string => {
+  const nonce = zeroPad(transaction.nonce, 2)
+  const gasLimit = zeroPad(transaction.gasLimit, 3)
+  const gasPrice = zeroPad(transaction.gasPrice, 1)
+  const to = hexStrToBuf(transaction.to)
+  const data = hexStrToBuf(transaction.data)
+
+  return Buffer.concat([
+    Buffer.from(nonce),
+    Buffer.from(gasLimit),
+    Buffer.from(gasPrice),
+    to,
+    data,
+  ]).toString('hex')
+}
+
+const encodeSequencerCalldata = async (
+  wallet: Wallet,
+  transaction: {
+    nonce: number
+    gasLimit: number
+    gasPrice: number
+    to: string
+    data: string
+    chainId?: number
+  },
+  transactionType: number
+) => {
+  transaction.chainId = 108
+
+  let serializedTransaction: string
+  if (transactionType === 2) {
+    serializedTransaction = ethers.utils.defaultAbiCoder.encode(
```

Comment:
I like the usage of `defaultAbiCoder.encode` more than my above implementation of `serializeEthSignTransaction`, we should create a function in `ovm-utils`